pywebostv/controls.py

```diff
@@ -119,6 +119,8 @@ def callback_wrapper(res):
             elif block:
                 res = self.request(cmd_info["uri"], params, block=block,
                                    timeout=timeout)
+                if res.get("type",None) == "error":
+                    raise IOError(res.get("error","Unknown Error(2)"))
```

Comment:
Thanks so much for this change :)

Can you also have a similar treatment before line 111 --except you'd say `callback(False, res.get("error", "..."))` ?

Reply:
yeah, would only be consequent of course. 
Haven't tried the non blocking at all. Will add it in and give you a feedback when I could test it!

Thanks for your work here in general and especially for your fast response! :)